src/utils/interpolate.hpp

```diff
@@ -0,0 +1,137 @@
+#ifndef PARTHENON_UTILS_INTERPOLATE_HPP_
+#define PARTHENON_UTILS_INTERPOLATE_HPP_
+//========================================================================================
+// Parthenon performance portable AMR framework
+// Copyright(C) 2022 The Parthenon collaboration
+// Licensed under the 3-clause BSD License, see LICENSE file for details
+//========================================================================================
+//! \file interpolate.hpp
+//  \brief Interpolation using cubic Hermite polynomials
+//
+// Performs monotone interpolation using cubic Hermite polynomials.
+// Reference: "Monotone piecewise cubic interpolation," SIAM J. Numer. Anal.,
+//            Vol. 17, No. 2, April 1980. [https://doi.org/10.1137/0717021]
+//========================================================================================
+
+#include "../kokkos_abstraction.hpp"
+#include <ostream>
+
+namespace parthenon {
+
+template <class VectorContainer>
```

Comment:
Can also add some concepts magic here for good measure
```suggestion
template<typename VectorContainer, class = ENABLEIF(implements<contiguous_container(VectorContainer)>::value)>
```
limits us to views with contiguous blocks of memory... but that's probably for the best.